### PR TITLE
use the updated mariadb10.6 container in drone

### PR DESCRIPTION
**Pull request by szaimen**
Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

**Comment by szaimen** [comment]
Cc @nickvergessen it seems like tests fail with thr latest mariadb10.6 version: `ERROR 1140 (42000) at line 2: Mixing of GROUP columns (MIN(),MAX(),COUNT(),...) with no GROUP columns is illegal if there is no GROUP BY clause`. What can we do to fix this?

I made it running by not adding the custom sql settings from the drone config in #33607 and #33606...

**Comment by nickvergessen** [comment]
> I made it running by not adding the sql-mode command from the drone config

Tests failed, so we don't run them.... 🤕

If the query is an issue it needs to be adjusted

**Comment by szaimen** [comment]
> If the query is an issue it needs to be adjusted

And where can I find this query and why does it fail only with this new mariadb10.6 version and not with an old mariadb10.6 version?

File: .drone.yml [hunk 1]
```diff
@@ -374,7 +374,7 @@ services:
 - name: cache
   image: ghcr.io/nextcloud/continuous-integration-redis:latest
 - name: mariadb
-  image: ghcr.io/nextcloud/continuous-integration-mariadb-10.6continuous-integration-mariadb-10.6:10.6
+  image: ghcr.io/nextcloud/continuous-integration-mariadb-10.6:latest
```
Comment (nickvergessen) [inline .drone.yml:377]: The diff looks weird, is that halfing intentional? Was it broken before? Or did maybe someone publish it under that name by accident but it's correct

**Comment by nickvergessen** [comment]
Maybe because they fixed something in a maintenance release?

I can't even see the query nor anything as CI just died with a timeout

File: .drone.yml (hunk 1 above)
Comment (szaimen) [reply .drone.yml:377]: > The diff looks weird, is that halfing intentional?

yes

> Was it broken before?

no

> Or did maybe someone publish it under that name by accident but it's correct

yes

But I published the new mariadb10.6 version under the name that I added here.

**Comment by szaimen** [comment]
> Maybe because they fixed something in a maintenance release?

possible

> I can't even see the query nor anything as CI just died with a timeout

same

**Comment by szaimen** [comment]
cc @dr-m maybe?

**Comment by dr-m** [comment]
@szaimen I only really know the InnoDB storage engine. On a quick read of the comments on this pull request, `sql_mode=ONLY_FULL_GROUP_BY` is what should cause the error to be reported for such invalid SQL. https://mariadb.com/kb/en/group-by/ does not say when the default setting of that might have been changed.

**Comment by szaimen** [comment]
> I only really know the InnoDB storage engine

Ah I see. Sorry for pinging you then!

> On a quick read of the comments on this pull request, `sql_mode=ONLY_FULL_GROUP_BY` is what should cause the error to be reported for such invalid SQL. https://mariadb.com/kb/en/group-by/ does not say when the default setting of that might have been changed.

Thanks! We'll have a look!

**Comment by szaimen** [comment]
Seems like removing `ONLY_FULL_GROUP_BY` indeed makes it work.

@nickvergessen since this was added intentionally, how can we debug and fix this?

**Comment by nickvergessen** [comment]
Maybe the docker image itself is incompatible with the flag? I don't know. At least our tests need to run in this mode, I don't really care how it gets enabled

**Comment by szaimen** [comment]
> Maybe the docker image itself is incompatible with the flag?

I'll try that out locally.

**Comment by szaimen** [comment]
Yes, this fails with the same log:
```
docker run -it --rm \
--name mariadb \
-e MYSQL_ROOT_PASSWORD=owncloud \
-e MYSQL_USER=oc_autotest \
-e MYSQL_PASSWORD=owncloud \
-e MYSQL_DATABASE=oc_autotest \
ghcr.io/nextcloud/continuous-integration-mariadb-10.6:latest \
--sql-mode=ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION
```

**Comment by szaimen** [comment]
Reported in https://jira.mariadb.org/browse/MDEV-29347

**Comment by szaimen** [comment]
Waiting for mariadb10.6.11 release which fixes the issue. https://hub.docker.com/_/mariadb/tags?page=1&name=10.6

**Comment by szaimen** [comment]
MariaDB 10.6.11 is finally out that brings the fix. So this should finally be ready for review :)

**Review by blizzz — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Comment by szaimen** [comment]
CI failure unrelated

**Pull request merged**